cypress/integration/teacher_adds_new_registered_student.spec.js

```diff
@@ -0,0 +1,115 @@
+import * as c from '../support/constants.js'
+import teacherHomePageElements from '../support/elements/teacher_home_page_elements.js'
+import addClassPageElements from '../support/elements/add_class_page_elements.js'
+import studentRosterPageElements from '../support/elements/student_roster_page_elements.js'
+import flashNoticePageElements from '../support/elements/flash_notice_page_elements.js'
+import manageClassesPageElements from '../support/elements/manage_classes_page_elements.js'
+import userHomePageElements from '../support/elements/user_home_page_elements.js'
+import changePasswordPageElements from '../support/elements/change_password_page_elements.js'
+import studentHomePageElements from '../support/elements/student_home_page_elements.js'
+
+// Note for db tracking : No db tracking required, using existing records
```

Comment:
This is adding the student to a class, and changing its password.
If one of the tests fails then the database will be left in an inconsistent state.
For testing locally or when running them on the CI (travis) this is fine because the database will be reset each time.

But I think you also want to run these tests on staging (and possibly production), so the smoke test can be automated. In that case we can't currently reset the database.

Reading the cypress docs on this: 
https://docs.cypress.io/guides/references/best-practices#Dangling-state-is-your-friend

It seems what we should do is clean up the database before the tests run. So we'll need to keep track of exactly what has been added, removed, or changed. Since the tests can fail we'll even need to know about records the cypress tests are cleaning up.

I guess these `note for db tracking` comments are the best way to document this, so we can implement that cleanup step later. I'm not sure yet what the best way to implement that cleanup step.